### ISSUE TITLE
Build failure: xdg-utils (aarch64-linux cross)

### ISSUE THREAD

**Issue by avnik** [NixOS/nixpkgs#292393]
### Steps To Reproduce

1. `nix-build -A pkgsCross.aarch64-multiplatform.xdg-utils -j1 --cores 16`

### Build log
A lot of similar failures via `binlore`/`yallback`/`resholve`.

```
building '/nix/store/7qdf9vjpizykfkwz7jfrdl9vvkp9d7sv-xdg-user-dirs-aarch64-unknown-linux-gnu-0.18-binlore.drv'...
generating binlore for /nix/store/7731h00maa974hx6hd1pa2cac14l3lii-xdg-user-dirs-aarch64-unknown-linux-gnu-0.18 by running:
/nix/store/g9cip59gy4vips6wfgnn8k0ds6ycw6ba-yara-aarch64-unknown-linux-gnu-4.4.0/bin/yara --scan-list --recursive /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yar <(printf '%s\n' /nix/store/7731h00maa974hx6hd1pa2cac14l3lii-xdg-user-dirs-aarch64-unknown-linux-gnu-0.18/{bin,lib,libexec}) | /nix/store/6f8b71wahxqn95013rrbji8j7x44zl0n-yallback-aarch64-unknown-linux-gnu-0.2.0/bin/yallback /nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yall
/build/.attr-0l2nkwhif96f51f4amnlf414lhl4rv9vh8iffyp431v6s28gsr90: line 21: /nix/store/g9cip59gy4vips6wfgnn8k0ds6ycw6ba-yara-aarch64-unknown-linux-gnu-4.4.0/bin/yara: cannot execute binary file: Exec format error
/nix/store/6f8b71wahxqn95013rrbji8j7x44zl0n-yallback-aarch64-unknown-linux-gnu-0.2.0/bin/.yallback-wrapped: line 4: shopt: progcomp: invalid shell option name
/nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yall: line 8: /nix/store/f1j59l21z68vr1ds4garm518czs328nc-coreutils-aarch64-unknown-linux-gnu-9.4/bin/mktemp: cannot execute binary file: Exec format error
/nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yall: line 8: /nix/store/f1j59l21z68vr1ds4garm518czs328nc-coreutils-aarch64-unknown-linux-gnu-9.4/bin/mktemp: cannot execute binary file: Exec format error
/nix/store/6f8b71wahxqn95013rrbji8j7x44zl0n-yallback-aarch64-unknown-linux-gnu-0.2.0/bin/.yallback-wrapped: line 31: compgen: command not found
/nix/store/6f8b71wahxqn95013rrbji8j7x44zl0n-yallback-aarch64-unknown-linux-gnu-0.2.0/bin/.yallback-wrapped: line 31: /nix/store/f1j59l21z68vr1ds4garm518czs328nc-coreutils-aarch64-unknown-linux-gnu-9.4/bin/sort: cannot execute binary file: Exec format error
/nix/store/6f8b71wahxqn95013rrbji8j7x44zl0n-yallback-aarch64-unknown-linux-gnu-0.2.0/bin/.yallback-wrapped: line 159: /nix/store/f1j59l21z68vr1ds4garm518czs328nc-coreutils-aarch64-unknown-linux-gnu-9.4/bin/sort: cannot execute binary file: Exec format error
/nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yall: line 54: /nix/store/f1j59l21z68vr1ds4garm518czs328nc-coreutils-aarch64-unknown-linux-gnu-9.4/bin/sort: cannot execute binary file: Exec format error
/nix/store/pj1yyp67mw43x0wib284ywgifll6rink-source/execers.yall: line 55: /nix/store/f1j59l21z68vr1ds4garm518czs328nc-coreutils-aarch64-unknown-linux-gnu-9.4/bin/sort: cannot execute binary file: Exec format error
note: keeping build directory '/var/buildroot/nix-build-xdg-user-dirs-aarch64-unknown-linux-gnu-0.18-binlore.drv-3'
```

### Additional context
1. `xdg-utils` depends on procmail, which also fail to cross-build
2. Following change fix some issues, but uncover more `resholve` stuff
```diff
diff --git a/pkgs/development/tools/analysis/binlore/default.nix b/pkgs/development/tools/analysis/binlore/default.nix
index 54ea108b7d46..ecc4399a4641 100644
--- a/pkgs/development/tools/analysis/binlore/default.nix
+++ b/pkgs/development/tools/analysis/binlore/default.nix
@@ -1,8 +1,7 @@
 { lib
 , fetchFromGitHub
 , runCommand
-, yallback
-, yara
+, buildPackages
 }:
 
 /* TODO/CAUTION:
@@ -27,6 +26,7 @@ that down will almost certainly mean reworking the API.
 */
 
 let
+  inherit (buildPackages) yara yallback;
   src = fetchFromGitHub {
     owner = "abathur";
     repo = "binlore";
```
### Notify maintainers

/cc @K900
also /cc @emilylange because it affecting chromium cross-build (sorry for bothering you)

### Metadata
Verified on  575147aed746e3230b46795176e50645b8fe8387 (origin/master, origin/HEAD)

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by danielbarter** [comment]
@abathur: 

With https://github.com/NixOS/nixpkgs/pull/298515 and https://github.com/NixOS/nixpkgs/pull/298001, I am still seeing issues with cross compiling `xdg-utils`:

```
Running phase: fixupPhase
[resholve context] : invoking resholve with PWD=/nix/store/b4dz07ykwb1vhvrpmwsn8q77rwvn4hg1-xdg-utils-aarch64-unknown-linux-gnu-1.2.1
[resholve context] RESHOLVE_LORE=/nix/store/1qz3ic8lv3sfdn42mdb99rsby6mma2ws-more-binlore
[resholve context] RESHOLVE_EXECER=cannot:/nix/store/icwsjb8xj0l6xaqiar55chysha6f0q8c-xdg-user-dirs-aarch64-unknown-linux-gnu-0.18/bin/xdg-user-dir
[resholve context] RESHOLVE_FAKE='external:'\''explorer.exe'\'';'\''gnome-default-applications-properties'\'';'\''kde-config'\'';'\''xprop'\'';'\''gconftool-2'\'''
[resholve context] RESHOLVE_INPUTS=/nix/store/11l9nr6di5dl98kncc128dq1c45dq4ha-dbus-aarch64-unknown-linux-gnu-1.14.10/bin:/nix/store/r1i320xwdhd7mkj38xhzakfir5nb1fda-coreutils-aarch64-unknown-linux-gnu-9.4/bin:/nix/store/vyw8s4fljapy9gklsgxlgsjxj5dsgfd1-gnugrep-aarch64-unknown-linux-gnu-3.11/bin:/nix/store/clmi4bhi11gs0qhikn26d8xm32dgbjm1-gnused-aarch64-unknown-linux-gnu-4.9/bin:/nix/store/icwsjb8xj0l6xaqiar55chysha6f0q8c-xdg-user-dirs-aarch64-unknown-linux-gnu-0.18/bin
[resholve context] RESHOLVE_INTERPRETER=/nix/store/xhz88dqwv29wkn3xnyin6qsh2lk48yd6-bash-aarch64-unknown-linux-gnu-5.2p26/bin/bash
[resholve context] RESHOLVE_KEEP='$KDE_SESSION_VERSION'
[resholve context] RESHOLVE_PROLOGUE=/nix/store/y5xn5jmgjkcj4ilswshxrfga21as2mj1-xdg-utils-prologue
[resholve context] /nix/store/g38al94ja8ks618x38why24zq68b265i-resholve-0.10.1/bin/resholve --overwrite bin/xdg-desktop-icon
                  eval 'cp "$desktop_file" "$x/$basefile"'$xdg_redirect_output
                       ^
/nix/store/b4dz07ykwb1vhvrpmwsn8q77rwvn4hg1-xdg-utils-aarch64-unknown-linux-gnu-1.2.1/bin/xdg-desktop-icon:781: FEEDBACK WANTED: Letting quoted 'eval' through for now.

https://github.com/abathur/resholve/issues/2
Traceback (most recent call last):
  File "/nix/store/g38al94ja8ks618x38why24zq68b265i-resholve-0.10.1/bin/.resholve-wrapped", line 4788, in <module>
    sys.exit(punshow())
  File "/nix/store/g38al94ja8ks618x38why24zq68b265i-resholve-0.10.1/bin/.resholve-wrapped", line 956, in punshow
    script_path, resolved_scripts[script_path], args.overwrite
  File "/nix/store/g38al94ja8ks618x38why24zq68b265i-resholve-0.10.1/bin/.resholve-wrapped", line 755, in write_resolved_script
    resolved.write_to(script_path)
  File "/nix/store/g38al94ja8ks618x38why24zq68b265i-resholve-0.10.1/bin/.resholve-wrapped", line 4540, in write_to
    self.render(to=f)
  File "/nix/store/g38al94ja8ks618x38why24zq68b265i-resholve-0.10.1/bin/.resholve-wrapped", line 4563, in render
    first_span = find_earliest_spid(self.root)
  File "/nix/store/g38al94ja8ks618x38why24zq68b265i-resholve-0.10.1/bin/.resholve-wrapped", line 1028, in find_earliest_spid
    sub_spid = find_earliest_spid(node.children[0])
  File "/nix/store/g38al94ja8ks618x38why24zq68b265i-resholve-0.10.1/bin/.resholve-wrapped", line 1023, in find_earliest_spid
    node_spid = node.keyword.span_id
AttributeError: 'NoneType' object has no attribute 'span_id'
/nix/store/df7g7q44kv8hv1n7nycigv4255n5whwy-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
```

**Comment by danielbarter** [comment]
This is still relevant. During configuration, some binaries are being compiled and run to decide things about the system. Not clear to me how to fix:

https://gist.github.com/danielbarter/a6a711e1f4f38368045c2dae2177ee7b

**Comment by uninsane** [comment]
fixed by #298515
